### PR TITLE
Add custom NACL rules for public and private subnets. Update versions and tests

### PR DESCRIPTION
**Pull request by aknysh**
## what
* Add custom NACL rules for public and private subnets
* Update module versions, GitHub workflows and tests
* Require Terraform version >= 1.3.0

## why
* Allow creating custom Network ACL rules for public and private subnets 
  - not all network architectures need ingress and egress rules on all ports (which this module can create as well)
  - custom NACL rules allow more fine-grained control over ingress and egress traffic

```hcl
# VPC CIDR block
ipv4_primary_cidr_block = "172.16.0.0/16"

# Create private subnets
private_subnets_enabled = true

# Create a Network ACL for the private subnets
private_network_acl_enabled = true

# Don't create all ingress and all egress rules for the private subnets
private_open_network_acl_enabled = false

# Create custom NACL rules for the private subnets
# https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/network_acl_rule
private_network_acl_rules = {
  "Allow TCP port 8080 from the same VPC" : {
    rule_action = "allow"
    rule_number = 10
    protocol    = "tcp"
    egress      = false
    cidr_block  = "172.16.0.0/16"
    from_port   = 8080
    to_port     = 8080
  }
  "Allow TCP port 22 from the same VPC" : {
    rule_action = "allow"
    rule_number = 20
    protocol    = "tcp"
    egress      = false
    cidr_block  = "172.16.0.0/16"
    from_port   = 22
    to_port     = 22
  }
}

# Create public subnets
public_subnets_enabled = true

# Create a Network ACL for the public subnets
public_network_acl_enabled = true

# Don't create all ingress and all egress rules for the public subnets
public_open_network_acl_enabled = false

# Create custom NACL rules for the public subnets
# https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/network_acl_rule
public_network_acl_rules = {
  "Allow all IPv4 egress" : {
    rule_action = "allow"
    rule_number = 10
    protocol    = "-1"
    egress      = true
    cidr_block  = "0.0.0.0/0"
    from_port   = 0
    to_port     = 0
  }
  "Allow all IPv6 egress" : {
    rule_action     = "allow"
    rule_number     = 20
    protocol        = "-1"
    egress          = true
    ipv6_cidr_block = "::/0"
    from_port       = 0
    to_port         = 0
  }
  "Allow TCP port 443 IPv4 ingress" : {
    rule_action = "allow"
    rule_number = 30
    protocol    = "tcp"
    egress      = false
    cidr_block  = "0.0.0.0/0"
    from_port   = 443
    to_port     = 443
  }
  "Allow TCP port 443 IPv6 ingress" : {
    rule_action     = "allow"
    rule_number     = 40
    protocol        = "tcp"
    egress          = false
    ipv6_cidr_block = "::/0"
    from_port       = 443
    to_port         = 443
  }
}
```

* Keep up to date

## related
* Closes https://github.com/cloudposse/terraform-aws-dynamic-subnets/issues/171

### CODE REVIEW

**Review by goruha — DISMISSED** [review]

**Review by Nuru — CHANGES_REQUESTED** [review]
I am opposed to adding inputs specifying NACL rules. This module can create the common patterns (single open NACL), but customization should be done by managing the NACLs outside of this module. (Either in the root module directly or via other modules referenced by the root module.)

Wrapping single resources like this has proven to be limiting for customers and overly burdensome for maintainers, and is not worth it in cases such as this where the module cannot provide meaningful simplifications and defaults.

cc @osterman

File: test/src/Makefile [hunk 1]
```diff
@@ -26,4 +26,4 @@ docker/test:
 .PHONY : clean
 ## Clean up files
 clean:
-	rm -rf $(TF_DATA_DIR) ../../examples/complete/*.tfstate*
+	rm -rf "$${TF_DATA_DIR}" ../../examples/complete/*.tfstate*
```
Comment (Nuru) [inline test/src/Makefile:29]: @aknysh this really makes me uncomfortable. If someone, somehow sets `TF_DATA_DIR="/"` then this is big trouble. Can you perform some kind of test to make sure `TF_DATA_DIR` points to something safe?

Comment (Nuru) [reply test/src/Makefile:29]: I don't see a reason for it to ever be in `test/src/Makefile`. If I put it there somewhere, please point it out to me, as I expect that was a mistake.

We sort of need it for the `clean` target at the top level Makefile for Geodesic, but really we should be using `atmos terraform clean` to make sure we get `TF_DATA_DIR` set, but then that doesn't work outside Geodesic.... smh

Comment (Nuru) [reply test/src/Makefile]: Confirmed, it is a mistake from converting from very old to current testing framework, propagated by me by mistake. See 
- https://github.com/cloudposse/terraform-aws-vpc-flow-logs-s3-bucket/pull/58
- https://github.com/cloudposse/terraform-aws-s3-bucket/pull/181

**Pull request closed without merging**